ScreenHandler -> Menu

Juuxel opened the pull request:
Replaces `ScreenHandler` with `Menu`, as seen in various registry and logging strings.

The name `Menu` is shown in unobfuscated names such as the `createMenu` method or the `minecraft:menu` registry (which Yarn incorrectly calls `SCREEN_HANDLER`).

This PR is the successor of #846. Also includes parts of #1166.

Juuxel committed:
ScreenHandler -> Menu

SuperCoder7979 commented:
how do i concern emote on github

liach commented on mappings/net/minecraft/client/gui/screen/ingame/AnvilScreen.mapping:
```diff
@@ -1,8 +1,8 @@
 CLASS net/minecraft/class_471 net/minecraft/client/gui/screen/ingame/AnvilScreen
-	FIELD field_2819 TEXTURE Lnet/minecraft/class_2960;
+	FIELD field_2819 BG_TEX Lnet/minecraft/class_2960;
```
```suggestion
	FIELD field_2819 TEXTURE Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/BeaconScreen.mapping:
```diff
@@ -3,9 +3,9 @@ CLASS net/minecraft/class_466 net/minecraft/client/gui/screen/ingame/BeaconScree
 	FIELD field_2804 doneButton Lnet/minecraft/class_466$class_468;
 	FIELD field_2805 consumeGem Z
-	FIELD field_2808 TEXTURE Lnet/minecraft/class_2960;
+	FIELD field_2808 BG_TEX Lnet/minecraft/class_2960;
```
```suggestion
	FIELD field_2808 TEXTURE Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/BrewingStandScreen.mapping:
```diff
@@ -2,6 +2,6 @@ CLASS net/minecraft/class_472 net/minecraft/client/gui/screen/ingame/BrewingStan
 	METHOD <init> (Lnet/minecraft/class_1708;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
-		ARG 2 inventory
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/CartographyTableScreen.mapping:
```diff
@@ -1,7 +1,7 @@
 	FIELD field_17421 TEXTURE Lnet/minecraft/class_2960;
 	METHOD <init> (Lnet/minecraft/class_3910;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/CraftingTableScreen.mapping:
```diff
@@ -1,9 +1,9 @@
-	FIELD field_2881 RECIPE_BUTTON_TEXTURE Lnet/minecraft/class_2960;
+	FIELD field_2878 BG_TEX Lnet/minecraft/class_2960;
+	FIELD field_2880 recipeBookGui Lnet/minecraft/class_507;
+	FIELD field_2881 RECIPE_BUTTON_TEX Lnet/minecraft/class_2960;
```
```suggestion
	FIELD field_2881 RECIPE_BUTTON_TEXTURE Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/CraftingTableScreen.mapping:
```diff
@@ -1,9 +1,9 @@
-	FIELD field_2878 TEXTURE Lnet/minecraft/class_2960;
-	FIELD field_2880 recipeBook Lnet/minecraft/class_507;
-	FIELD field_2881 RECIPE_BUTTON_TEXTURE Lnet/minecraft/class_2960;
+	FIELD field_2878 BG_TEX Lnet/minecraft/class_2960;
```
```suggestion
	FIELD field_2878 TEXTURE Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/ForgingContainerScreen.mapping:
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_4894 net/minecraft/client/gui/screen/ingame/ForgingContainerScreen
```
Should be `ForgingMenuScreen`

liach commented on mappings/net/minecraft/client/gui/screen/ingame/GrindstoneScreen.mapping:
```diff
@@ -1,6 +1,6 @@
 	FIELD field_16769 TEXTURE Lnet/minecraft/class_2960;
 	METHOD <init> (Lnet/minecraft/class_3803;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/HopperScreen.mapping:
```diff
@@ -1,6 +1,6 @@
 	FIELD field_2919 TEXTURE Lnet/minecraft/class_2960;
 	METHOD <init> (Lnet/minecraft/class_1722;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/InventoryScreen.mapping:
```diff
@@ -3,7 +3,7 @@ CLASS net/minecraft/class_490 net/minecraft/client/gui/screen/ingame/InventorySc
 	FIELD field_2931 isNarrow Z
 	FIELD field_2932 isOpen Z
-	FIELD field_2933 RECIPE_BUTTON_TEXTURE Lnet/minecraft/class_2960;
+	FIELD field_2933 RECIPE_BUTTON_TEX Lnet/minecraft/class_2960;
```
```suggestion
	FIELD field_2933 RECIPE_BUTTON_TEXTURE Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/HorseScreen.mapping:
```diff
@@ -4,6 +4,6 @@ CLASS net/minecraft/class_491 net/minecraft/client/gui/screen/ingame/HorseScreen
 	FIELD field_2941 entity Lnet/minecraft/class_1496;
 	METHOD <init> (Lnet/minecraft/class_1724;Lnet/minecraft/class_1661;Lnet/minecraft/class_1496;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/LoomScreen.mapping:
```diff
@@ -11,7 +11,7 @@ CLASS net/minecraft/class_494 net/minecraft/client/gui/screen/ingame/LoomScreen
 	FIELD field_2970 firstPatternButtonId I
 	METHOD <init> (Lnet/minecraft/class_1726;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/MerchantScreen.mapping:
```diff
@@ -5,7 +5,7 @@ CLASS net/minecraft/class_492 net/minecraft/client/gui/screen/ingame/MerchantScr
 	FIELD field_2950 TEXTURE Lnet/minecraft/class_2960;
 	METHOD <init> (Lnet/minecraft/class_1728;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/ShulkerBoxScreen.mapping:
```diff
@@ -1,6 +1,6 @@
 	FIELD field_2973 TEXTURE Lnet/minecraft/class_2960;
 	METHOD <init> (Lnet/minecraft/class_1733;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/SmithingContainerScreen.mapping:
```diff
@@ -1,6 +1,6 @@
-CLASS net/minecraft/class_4895 net/minecraft/client/gui/screen/ingame/SmithingScreen
+CLASS net/minecraft/class_4895 net/minecraft/client/gui/screen/ingame/SmithingContainerScreen
```
Should be `SmithingMenuScreen`

liach commented on mappings/net/minecraft/client/gui/screen/ingame/SmithingContainerScreen.mapping:
```diff
@@ -1,6 +1,6 @@
 	FIELD field_22795 TEXTURE Lnet/minecraft/class_2960;
 	METHOD <init> (Lnet/minecraft/class_4862;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/client/gui/screen/ingame/StonecutterScreen.mapping:
```diff
@@ -5,7 +5,7 @@ CLASS net/minecraft/class_3979 net/minecraft/client/gui/screen/ingame/Stonecutte
 	FIELD field_17674 scrollAmount F
 	METHOD <init> (Lnet/minecraft/class_3971;Lnet/minecraft/class_1661;Lnet/minecraft/class_2561;)V
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach commented on mappings/net/minecraft/data/client/model/ModelIds.mapping:
```diff
@@ -9,7 +9,7 @@ CLASS net/minecraft/class_4941 net/minecraft/data/client/model/ModelIds
 		ARG 0 block
 		ARG 1 suffix
-	METHOD method_25844 getMinecraftNamespacedBlock (Ljava/lang/String;)Lnet/minecraft/class_2960;
+	METHOD method_25844 getMinecraftBlock (Ljava/lang/String;)Lnet/minecraft/class_2960;
```
```suggestion
	METHOD method_25844 getMinecraftNamespacedBlock (Ljava/lang/String;)Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/data/client/model/ModelIds.mapping:
```diff
@@ -9,7 +9,7 @@ CLASS net/minecraft/class_4941 net/minecraft/data/client/model/ModelIds
+	METHOD method_25844 getMinecraftBlock (Ljava/lang/String;)Lnet/minecraft/class_2960;
 		ARG 0 name
-	METHOD method_25845 getMinecraftNamespacedItem (Ljava/lang/String;)Lnet/minecraft/class_2960;
+	METHOD method_25845 getMinecraftItem (Ljava/lang/String;)Lnet/minecraft/class_2960;
```
```suggestion
	METHOD method_25845 getMinecraftNamespacedItem (Ljava/lang/String;)Lnet/minecraft/class_2960;
```

liach commented on mappings/net/minecraft/entity/player/PlayerEntity.mapping:
```diff
@@ -51,8 +51,7 @@ CLASS net/minecraft/class_1657 net/minecraft/entity/player/PlayerEntity
 		ARG 6 refreshable
-	METHOD method_17355 openHandledScreen (Lnet/minecraft/class_3908;)Ljava/util/OptionalInt;
-		ARG 1 factory
+	METHOD method_17355 openMenu (Lnet/minecraft/class_3908;)Ljava/util/OptionalInt;
```
```suggestion
	METHOD method_17355 openMenu (Lnet/minecraft/class_3908;)Ljava/util/OptionalInt;
		ARG 1 factory
```

liach commented on mappings/net/minecraft/menu/AnvilMenu.mapping:
```diff
@@ -1,4 +1,5 @@
-CLASS net/minecraft/class_1706 net/minecraft/screen/AnvilScreenHandler
+CLASS net/minecraft/class_1706 net/minecraft/menu/AnvilMenu
+	FIELD field_7769 result Lnet/minecraft/class_1263;
```
This field should've been gone
```suggestion
```

liach commented on mappings/net/minecraft/menu/MenuListener.mapping:
```diff
@@ -0,0 +1,12 @@
+	METHOD method_7634 onMenuRegistered (Lnet/minecraft/class_1703;Lnet/minecraft/class_2371;)V
+		ARG 1 menu
+		ARG 2 stacks
+	METHOD method_7635 onMenuSlotUpdate (Lnet/minecraft/class_1703;ILnet/minecraft/class_1799;)V
```
Don't think these methods need `Menu` in their names

liach commented on mappings/net/minecraft/menu/NameableMenuFactory.mapping:
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_3908 net/minecraft/menu/NameableMenuFactory
+	METHOD method_5476 getDisplayName ()Lnet/minecraft/class_2561;
```
Should be like `getTitle`

Juuxel replied:
I'm going to revert that change since it's the same obf/intermediary method as `Nameable.getDisplayName` which is used by entities like players.

liach commented on mappings/net/minecraft/menu/SmithingTableMenu.mapping:
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_4862 net/minecraft/screen/SmithingScreenHandler
+CLASS net/minecraft/class_4862 net/minecraft/menu/SmithingTableMenu
```
Should be smithing menu as the menu's id is `minecraft:smithing`

Ueaj-Kerman reviewed (commented):
Why

liach commented on mappings/net/minecraft/network/packet/c2s/play/ConfirmGuiActionC2SPacket.mapping:
```diff
@@ -3,8 +3,8 @@ CLASS net/minecraft/class_2809 net/minecraft/network/packet/c2s/play/ConfirmGuiA
 	FIELD field_12811 windowId I
 	METHOD <init> (ISZ)V
-		ARG 1 windowId
+		ARG 1 syncId
```
The packet should probably be renamed to "confirm menu action"

liach commented on mappings/net/minecraft/network/packet/s2c/play/ConfirmGuiActionS2CPacket.mapping:
```diff
@@ -1,11 +1,11 @@
 CLASS net/minecraft/class_2644 net/minecraft/network/packet/s2c/play/ConfirmGuiActionS2CPacket
```
Should be "confirm menu action"

liach commented on mappings/net/minecraft/recipe/book/RecipeBook.mapping:
```diff
@@ -20,7 +20,7 @@ CLASS net/minecraft/class_3439 net/minecraft/recipe/book/RecipeBook
 		ARG 1 id
 	METHOD method_14880 isFilteringCraftable (Lnet/minecraft/class_1729;)Z
-		ARG 1 handler
+		ARG 1 container
```
```suggestion
		ARG 1 menu
```

liach reviewed (changes requested):
Quite a few places where you've reverted `handler` to `container`.

modmuss50 reviewed (changes requested):
I dont see this being merged in, we just spent months coming to an agreement on a replacement to Container.

We won't be changing names like this every day.

https://github.com/FabricMC/yarn/pull/1106 Was mostly agreed upon to not be bad, it had no outstanding reviews.

florensie commented:
Phew. I was just about to say I'd go commit die if it were to get merged.

Juuxel committed:
Remove pointless/incorrect changes, slot cleanup

Juuxel committed:
Apply suggestions from code review

Co-Authored-By: liach <7806504+liach@users.noreply.github.com>

Juuxel commented:
> #1106 Was mostly agreed upon to not be bad

It creates a bunch of nonsense names like `ScreenHandler createMenu();` or `Registry<...> SCREEN_HANDLER = create("menu")`.

modmuss50 commented:
We were aware of those 2 places, and agreed menu wasn’t the right for us.

Juuxel committed:
Some refactoring

Juuxel committed:
Update CONVENTIONS.md

Juuxel committed:
NameableMenuFactory.getTitle() -> getDisplayName()

It's the same method as Nameable.getDisplayName() which is also used
by entities such as players. (I don't think players have a title? :P)

Juuxel committed:
Port some changes from #1166

Co-authored-by: liach <liach@users.noreply.github.com>

Juuxel committed:
Restore comments

Juuxel committed:
Remove some menu prefixes from MenuListener methods

Juuxel committed:
Merge branch '20w11a' into new-menu

modmuss50 reviewed (changes requested):
Just so you know, I dont see this been merged. A lot of discussion happened over a long period of time and it was clear the `Menu` was not the right choice for yarn.

Feel free to maintain your own fork that has these changes, but we wont be making large yarn changes every other day driving mod devs nuts.

Juuxel commented:
Yes, I know this most likely won't ever get merged. I will still maintain this for my own use, though, as `ScreenHandler` isn't the best name ever. I wish we'd gone with `ScreenController`, but what can I do  ¯\\\_(ツ)\_/¯

liach commented:
@Juuxel so can I close this pr so I am less concerned with the number of open prs in yarn... I want things clean

Juuxel committed:
Merge remote-tracking branch 'upstream/20w12a' into new-menu

Yoghurt4C reviewed (commented):
based

Juuxel committed:
Some renames to be more consistent with upstream

Juuxel committed:
Merge branch '20w14a' into new-menu

valoeghese commented:
screenhandler bad

shedaniel commented:
i have to defend from 2 words names, back to container bois

valoeghese commented:
screenhandler bad

kanpov commented:
this is forever going to be a part of history
